evalml/pipelines/components/transformers/imputers/target_imputer.py

```diff
@@ -0,0 +1,95 @@
+import pandas as pd
+import woodwork as ww
+from sklearn.impute import SimpleImputer as SkImputer
+
+from evalml.pipelines.components.transformers import Transformer
+from evalml.utils import (
+    _convert_woodwork_types_wrapper,
+    _retain_custom_types_and_initalize_woodwork,
+    infer_feature_types
+)
+
+
+class TargetImputer(Transformer):
```

Comment:
@angela97lin Did you try subclassing the evalml `SimpleImputer` or using composition? Seems like we're duplicating the implementation, except that this component only looks at y.

Reply:
I did briefly, but there wasn't too much codeshare (since different checks for 1d vs 2d) and I liked having the methods in this class explicitly for the docstrings too so I ended deciding against it, but I'm down to whip it back up--maybe I overlooked some places that could be cleaned up 😅

Reply:
Heh, looks like `_get_subclasses` breaks because it only expects "leaf nodes" and by subclassing SimpleImputer, TargetImputer becomes a leaf node and SimpleImputer is no longer an acceptable subclass. I think I'm going to punt on subclassing `SimpleImputer` for now and file an issue for this :d

Reply:
@angela97lin Sounds good! Thanks for looking into this!

Reply:
Filed #2021 (Year 2021, hey!), thank you for suggesting this @freddyaboulton!